crates/ruff_python/src/string.rs

```diff
@@ -3,7 +3,8 @@ use regex::Regex;
 
 pub static STRING_QUOTE_PREFIX_REGEX: Lazy<Regex> =
     Lazy::new(|| Regex::new(r#"^(?i)[urb]*['"](?P<raw>.*)['"]$"#).unwrap());
-pub static LOWER_OR_UNDERSCORE: Lazy<Regex> = Lazy::new(|| Regex::new(r"^[a-z_]+$").unwrap());
+pub static LOWER_OR_UNDERSCORE: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"^[a-z][a-z0-9_]*$").unwrap());
```

Comment:
Why the different regex than in `flake8_module_name`?

https://github.com/Ohjeah/flake8_module_name/blob/master/flake8_module_name.py#L6

It now rejects module name `_module` which was not the case before.

PEP8 discourages underscores in module names but also states:
> Even with __all__ set appropriately, internal interfaces (packages, modules, classes, functions, attributes or other names) should still be prefixed with a single leading underscore.

https://peps.python.org/pep-0008/#public-and-internal-interfaces

Can we reconsider changing this?

Reply:
Ah yeah, this seems off. @sbrugman, what do you think?

Reply:
also `0001_initial.py` is now invalid, but works with flake-module-name

Reply:
👍 Same issue. I agree that we should change this but want to see if @sbrugman deviated for a reason.

Reply:
I will take a look at this now and perhaps cut a new release real quick.

Reply:
https://github.com/charliermarsh/ruff/pull/3043

Reply:
Thanks for fixing!